### PR TITLE
Reactive Oracle Client extension

### PR DESCRIPTION
**Pull request by tsegismont**
See #21432

This PR adds an extension for the Vert.x Reactive Oracle Client.

There are integrations tests that pass both in JVM and native mode.
However, native mode has not been tested extensively.

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

**Comment by geoand** [comment]
CI is not happy. See the logs for details on how to address the situation

**Comment by tsegismont** [comment]
@geoand ran `./update-extension-dependencies.sh` and now :crossed_fingers:

**Review by gsmet — CHANGES_REQUESTED** [review]
Make sure you squash everything once CI is green. Thanks.

**Comment by gsmet** [comment]
And btw, that's a great addition, thanks!

**Comment by geoand** [comment]
Also, it's probably best to squash the commits

**Comment by tsegismont** [comment]
@geoand @gsmet I looked at the failures and they come from missing `quarkus-amazon-*` artifacts. Does that ring a bell?

**Comment by geoand** [comment]
I personally don't remember seeing that failure. Let me restart CI

**Comment by gsmet** [comment]
The changes made to `devtools/bom-descriptor-json/pom.xml` are incorrect. They are referencing artifacts that have been moved out of the tree.

**Comment by tsegismont** [comment]
Thank you Guillaume. I had rebased my on main before sending the PR. I will
do this again.

**Comment by tsegismont** [comment]
@gsmet @geoandri I just rebased on main and ran `./update-extension-dependencies.sh` again. No file has been modified.

I can modify `devtools/bom-descriptor-json/pom.xml` manually but it seems there's a problem with the script. Do you want me to file an issue?

**Comment by famod** [comment]
Let me have a quick look...

**Comment by famod** [comment]
@tsegismont 
> Do you want me to file an issue?

Yes, please assign it to me.
It seems there is an inconsistency in how `devtools\bom-descriptor-json\target\quarkus-bom-quarkus-platform-descriptor-999-SNAPSHOT-999-SNAPSHOT.json` is build when there are still removed extensions in local repo. /cc @aloubyansky

**Comment by aloubyansky** [comment]
What do you mean by "still removed extensions in local repo" @famod? That they haven't been installed yet?

**Comment by famod** [comment]
Problem confirmed: As long as you still have those extensions installed in local repo that were removed via #21736, `quarkus-bom-quarkus-platform-descriptor-999-SNAPSHOT-999-SNAPSHOT.json` will still include those.
The problem vanishes after `find ~/.m2/repository/io/quarkus -type d -name 999-SNAPSHOT -exec rm -rf {} \;`.

So _everyone_ that has cloned and built this repo before #21736 will run into this issue.
CI doesn't have this problem because it always starts without any `io.quarkus` artifacts in local repo.

**Comment by aloubyansky** [comment]
Even if you build with `-Prelocations`?

**Comment by famod** [comment]
Haven't tried this, but it's not supposed to be used by default, or is it?

**Comment by aloubyansky** [comment]
Maybe it should be.

**Comment by famod** [comment]
But wasn't the idea to _reduce_ the modules? But sure, such a relocation is quick to build.

Feel free to try it, I have to stop here for now (and those old extensions are now gone from my local repo).

Just check ` grep amazon-kms devtools/bom-descriptor-json/target/quarkus-bom-quarkus-platform-descriptor-*.json` after a full build, there should be matches (but it shouldn't be there).
Rebuild after the removal from local repo and the there shouldn't be a match anymore.

**Comment by famod** [comment]
This ⬆️ is all on `main` btw!

**Comment by tsegismont** [comment]
@famod https://github.com/quarkusio/quarkus/issues/22836

File: extensions/reactive-oracle-client/deployment/pom.xml [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    </parent>
+
+    <artifactId>quarkus-reactive-oracle-client-deployment</artifactId>
+
+    <name>Quarkus - Reactive Oracle Client - Deployment</name>
+
+    <properties>
+        <reactive-oracledb.url>vertx-reactive:oracle:thin:@localhost:1521:XE</reactive-oracledb.url>
+        <!-- See https://hub.docker.com/r/gvenzl/oracle-xe -->
+        <oracle.image>gvenzl/oracle-xe:21.3.0-slim</oracle.image>
```
Comment (gsmet) [inline extensions/reactive-oracle-client/deployment/pom.xml:20]: I think we are already using this image somewhere else so maybe we should centralize it in the `build-parent`? But that can probably be done in a follow-up PR.

/cc @Sanne

**Review by gsmet — APPROVED** [review]
Looks good to me. I added a small non-blocking comment and I think it should be done as a follow-up and not in this PR.

File: extensions/reactive-oracle-client/deployment/pom.xml (hunk 1 above)
Comment (Sanne) [reply extensions/reactive-oracle-client/deployment/pom.xml:20]: Right - we use it in `integration-tests/jpa-oracle/pom.xml` as well.

+1 to follow up later, not really a blocker.

**Review by cescoffier — APPROVED** [review]
Awesome! Thanks!

File: extensions/reactive-oracle-client/deployment/pom.xml (hunk 1 above)
Comment (tsegismont) [reply extensions/reactive-oracle-client/deployment/pom.xml:20]: Will do

**Comment by tsegismont** [comment]
@gsmet @cescoffier given the Reactive Oracle Client is fundamentally different from the others, how can we tell users that the extension shall be used for evaluation purpose only and not yet for production?

**Comment by aloubyansky** [comment]
Like this https://github.com/quarkusio/quarkus/blob/main/extensions/security/runtime/src/main/resources/META-INF/quarkus-extension.yaml#L10

**Comment by tsegismont** [comment]
OK, thanks. Then we're good already
https://github.com/quarkusio/quarkus/blob/main/extensions/reactive-oracle-client/runtime/src/main/resources/META-INF/quarkus-extension.yaml#L17

**Comment by gsmet** [comment]
@tsegismont I think a small note in the documentation could be useful too. You can use this as an inspiration: https://raw.githubusercontent.com/quarkusio/quarkus/main/docs/src/main/asciidoc/status-include.adoc (but just put the text at the right place, no need to use the include).

**Comment by tsegismont** [comment]
@IoT-Hub-RU hi, I've considered it but decided to skip the `jdbc` prefix for consistency with other Reactive SQL clients and to avoid confusion with Vert.x JDBC Client.

**Pull request merged**